feat: Public org profile page

SuaYoo opened the pull request:
Resolves #2156
Resolves #2133
Follows https://github.com/webrecorder/browsertrix/pull/2174 (must be deployed to dev for manual testing steps)

## Changes

- Enables creating a public org profile page with description and website at `/profile/<org slug>`
- Updates current "Overview" page to be "Dashboard", found under `/dashboard`
- Organizes org "General" settings tab by "General", "Profile", and "Developer Tools"
- Adds sign up banner to log in page for consistent CTA banners
- Updates copy and docs to support changes

## Manual testing

1. Log in as crawler
2. Go to org "Settings"
3. Update profile description and website. Verify values save as expected
4. Click "Preview public profile page". Verify banner is shown with text describing that profile is still private
5. Log out
6. Go to `/profile/<your org slug>`. Verify generic "not found" message is shown
7. Log back in
8. Go back to org settings
9. Enable "Allow anyone to see org"
10. Go back to public profile page and refresh. Verify preview banner is no longer shown, while controls to edit the page are still visible
11. Log out
12. Go to `/profile/<your org slug>`. Verify page is still visible

## Screenshots

### Org Settings
<img width="1028" alt="Screenshot 2024-11-26 at 2 35 23 PM" src="https://github.com/user-attachments/assets/3ac0369f-eac4-409e-82c7-05a75a5d1b61">

<img width="1030" alt="Screenshot 2024-11-26 at 2 35 46 PM" src="https://github.com/user-attachments/assets/779d7a59-827f-4d5f-af5f-e80973dc1f0d">

<img width="1032" alt="Screenshot 2024-11-26 at 2 35 52 PM" src="https://github.com/user-attachments/assets/ad85a9b3-c1df-47e5-9e3b-654cb241288e">

### Org Profile (private preview)

<img width="794" alt="Screenshot 2024-11-26 at 2 57 36 PM" src="https://github.com/user-attachments/assets/e8cbe0dd-f0f4-4abd-bff6-1741c9dc42c7">


### Org Profile (public)
<img width="802" alt="Screenshot 2024-11-26 at 2 57 54 PM" src="https://github.com/user-attachments/assets/d293bc48-5ab1-43f8-9f92-9a0fef485a54">



### Sign In
<img width="802" alt="Screenshot 2024-11-26 at 2 56 43 PM" src="https://github.com/user-attachments/assets/7228dcce-176a-424b-a4df-2499d9ba7cd7">




### Generic 404

<img width="797" alt="Screenshot 2024-11-26 at 2 56 54 PM" src="https://github.com/user-attachments/assets/b40245b9-4cd7-45b5-9129-fd79020ae248">


### Docs: User Guide > Org Settings > Change Org Settings
<img width="729" alt="Screenshot 2024-11-26 at 3 29 39 PM" src="https://github.com/user-attachments/assets/5beb2e9e-4dbe-4010-a37f-1114b00154a6">

## Follow-ups

This change doesn't include actual public collections yet, which will be addressed with https://github.com/webrecorder/browsertrix/issues/2182

SuaYoo committed:
update type

SuaYoo committed:
add section

SuaYoo committed:
separate dashboard from home

SuaYoo committed:
update not found

SuaYoo committed:
add placeholder collections

SuaYoo committed:
update logged out view

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
tweak login

SuaYoo committed:
add cta

SuaYoo committed:
update public page

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
hide cta if logged in

SuaYoo committed:
show port

SuaYoo committed:
update org name description

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
update control

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
update field name

SuaYoo committed:
update tests

SuaYoo committed:
add routes enum

SuaYoo committed:
handle org slug

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
add description

SuaYoo committed:
update profile

SuaYoo committed:
update sign up link

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
update copy

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
update docs

tw4l commented:
Merged public collections API branch into feature branch and rebased this on that so it's now possible to run backend + frontend locally off this branch.

SuaYoo committed:
fix org slug in state on login

Shrinks99 commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -0,0 +1,158 @@
+        `,
+        html`
+          ${msg(
+            "To customize this URL, update your Org URL in General settings.",
```
I don't see a lot of value to this field in this configuration.  It seems like duplicate content as profile page is right above.  Could we add the copy button there to retain that functionality and remove the separate view-only field to display the URL?

Shrinks99 commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -0,0 +1,158 @@
+          ></sl-textarea>
+        `,
+        msg(
+          "Write a short description that introduces your organization or your web archive.",
```
Would remove "or your web archive" to avoid confusion with collection descriptions?

```suggestion
          "Write a short description to introduce your organization.",
```

Shrinks99 commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -0,0 +1,158 @@
+            label=${msg("Website")}
+            value=${this.org?.publicUrl || ""}
+            minlength="2"
+            placeholder="https://"
```
Can we auto-prepend `https://` if the user leaves it out?  This is a frequent cause for confusion on other fields, users aren't used to websites being as strict as we are and not fixing it automatically.

```suggestion
            placeholder="https://example.com"
```

Additionally / or, we need some error text telling people that the required format is `https://example.com` below the field so when it is in its error state they have more information on a resolution.

tw4l replied:
Ilya just added auto-prepending https in the workflow editor, could crib the solution from that: https://github.com/webrecorder/browsertrix/pull/2177

Shrinks99 commented on frontend/src/pages/log-in.ts:
```diff
@@ -215,15 +215,31 @@ export class LogInPage extends BtrixElement {
+            <div
+              class="w-full gap-4 border-y bg-white/30 p-6 px-3 text-center text-neutral-500"
+            >
+              <span>${msg("Need an account?")}</span>
```
Nice!  Linking #2133 to close that issue with this PR because I think this addresses it?  @emma-sg if you have any additional thoughts?

Shrinks99 committed:
Make list more concise

Shrinks99 reviewed (commented):
A great start to the future of public content on Browsertrix!!

SuaYoo committed:
namespace orgs

SuaYoo committed:
update docs

SuaYoo committed:
update website field

SuaYoo committed:
Update frontend/src/pages/org/settings/components/profile.ts

Co-authored-by: Henry Wilkinson <henry@wilkinson.graphics>

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
rename to profile

SuaYoo commented on frontend/src/routes.ts:
```diff
@@ -1,3 +1,17 @@
+
+export enum RouteNamespace {
+  PrivateOrgs = "orgs",
+  PublicOrgs = "profile",
```
This can easily be changed if someone has a pithy and creative idea for a public namespace. I want us to consider something other than `public` since that's helpful to us internally by being quite literal, but to anyone who's able to view the pages it'll be redundant, since they can see the page.

SuaYoo committed:
fix test

emma-sg committed:
remove padding between profile page input and prefix

emma-sg committed:
add a todo for scrolling to org url input

emma-sg commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -0,0 +1,159 @@
+              size="small"
+              ?checked=${this.org?.enablePublicProfile}
+            >
+              ${msg("Allow anyone to see org")}
```
I feel like this might be better off as a select element — but barring that, maybe this helps the label feel a little better?
```suggestion
              ${msg("Allow anyone to view org")}
```

emma-sg commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -0,0 +1,159 @@
+          </div>
+        `,
+        msg(
+          "If enabled, anyone with the link to your org's Browsertrix URL will be able to view the profile page and public collections.",
```
```suggestion
          "If enabled, anyone will be able to view your org's profile page and public collections.",
```

emma-sg reviewed (commented):
Just a couple thoughts, but this looks great otherwise!

SuaYoo committed:
Apply suggestions from code review

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

SuaYoo committed:
Apply `localize:extract` changes

emma-sg committed:
Verified badges (#2195)

Adds a simple verified badge to public pages (off by default currently
until there's a corresponding backend change).

To be merged after #2172 (& once corresponding backend changes are
done).

<img width="1057" alt="Screenshot_2024-12-02_at_3 12 57_PM"
src="https://github.com/user-attachments/assets/4a70b7c8-9c88-42c3-ae78-59a12228f277">
<img width="318" alt="Screenshot_2024-12-02_at_3 13 54_PM"
src="https://github.com/user-attachments/assets/15985c2e-9ac7-4d65-a668-578807487766">

---------

Co-authored-by: emma-sg <emma-sg@users.noreply.github.com>

SuaYoo commented on frontend/src/pages/org/settings/components/profile.ts:
```diff
@@ -80,6 +78,12 @@ export class OrgSettingsProfile extends BtrixElement {
+            // as anchor links don't penetrate the shadow dom. Instead we'd most likely
+            // need to set up a global shared re-implementation of the id system, with
+            // a globally-shared map with `@lit/context` or `lit-shared-state`. Another
+            // small frustration about web components!
```
@emma-sg This could be a link to `#org-url` and `btrix-org-settings` could scroll to the correct section on a hash change event, or did you have something else in mind?

emma-sg committed:
link to org url field

emma-sg committed:
Apply `localize:extract` changes
